packages/react/src/components/uiComponents.tsx

```diff
@@ -248,20 +189,37 @@ export function UserProfileLink({ children }: PropsWithChildren<UserProfileLinkP
 }
 
 const _UserProfile = withClerk(
-  ({ clerk, ...props }: WithClerkProp<PropsWithChildren<Without<UserProfileProps, 'customPages'>>>) => {
+  ({
+    clerk,
+    component,
+    fallback,
+    ...props
+  }: WithClerkProp<PropsWithChildren<Without<UserProfileProps, 'customPages'>> & FallbackProp>) => {
+    const mountingStatus = useWaitForComponentMount(component);
+    const shouldShowFallback = mountingStatus === 'rendering' || !clerk.loaded;
+
+    const rendererRootProps = {
+      ...(shouldShowFallback && fallback && { style: { display: 'none' } }),
+    };
+
     const { customPages, customPagesPortals } = useUserProfileCustomPages(props.children);
     return (
-      <Portal
-        mount={clerk.mountUserProfile}
-        unmount={clerk.unmountUserProfile}
-        updateProps={(clerk as any).__unstable__updateProps}
-        props={{ ...props, customPages }}
-      >
-        <CustomPortalsRenderer customPagesPortals={customPagesPortals} />
-      </Portal>
+      <>
+        {shouldShowFallback && fallback}
+        <ClerkHostRenderer
+          component={component}
+          mount={clerk.mountUserProfile}
+          unmount={clerk.unmountUserProfile}
+          updateProps={(clerk as any).__unstable__updateProps}
+          props={{ ...props, customPages }}
+          rootProps={rendererRootProps}
+        >
+          <CustomPortalsRenderer customPagesPortals={customPagesPortals} />
+        </ClerkHostRenderer>
+      </>
     );
   },
-  'UserProfile',
+  { component: 'UserProfile', renderWhileLoading: true },
```

Comment:
In which scenario do we want this to be `false` ?

Reply:
The existing behavior is for `withClerk()` to not render its children until Clerk has loaded. We use it for most other components, so this lets us retain existing behavior without making a larger refactor.

I do think it's worth dropping the option in the future, we'll just need to update all usages 👍